Makefile

```diff
@@ -144,6 +144,34 @@ vet-all: crates-vet modules-vet
 ci-checks: lint-all test-all vet-all
 	@echo "All CI checks completed successfully."
 
+# checking if the test dependencies are installed
+check-deps:
+	@command -v k3d >/dev/null 2>&1 || { echo "❌ 'k3d' is not installed, Please install it from https://k3d.io/stable/#releases to run the tests"; exit 1; }
+	@command -v kubectl >/dev/null 2>&1 || { echo "❌ 'kubectl' is not installed, Please install it from https://kubernetes.io/docs/tasks/tools to run the tests"; exit 1; }
+	@echo "✅ All dependencies are installed"
+
+# Testing if we can read the expected logs with kubetail 
+test:check-deps build
+	@if ! k3d cluster list | grep -q 'kubetail-test-cluster'; then \
+		echo "🛠️ Cluster not found. Creating..."; \
+		k3d cluster create kubetail-test-cluster; \
+	else \
+		echo "✅ Cluster already exists. Skipping creation."; \
+	fi
+	
+	@kubectl wait --for=condition=Ready nodes --all --timeout=60s
+	@kubectl apply -f .github/ci-config/k3d/log-demo.yaml
```

Comment:
Since we moved the business logic of the e2e-tests to `make`, I don't think it makes sense to keep the workload manifest in `.github/ci-config` anymore. Let's put the manifest in a new `hack/e2e-test` directory (or a more appropriate name if you can think of one).

Reply:
Also, let's come up with a different name than `log-demo.yaml`.